### PR TITLE
:sparkles: feat: support openrouter custom models env

### PR DESCRIPTION
**Pull request by danielglh**
#### 💻 变更类型 | Change Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] ⚡️ perf
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

Allow `OPENROUTER_CUSTOM_MODELS` env var to define custom models for OpenRouter provider, together with its provider setting. It works similarly to the `CUSTOM_MODELS` env var for OpenAI:

image

Fixed issue https://github.com/lobehub/lobe-chat/issues/1624.

#### 📝 补充信息 | Additional Information

N/A

### CODE REVIEW

**Comment by arvinxx** [comment]
The implement is LGTM. Could you please add a test for the model selector to cover openrouter env logic?

**Comment by danielglh** [comment]
> The implement is LGTM. Could you please add a test for the modelSelectList selector to cover openrouter env logic?

@arvinxx I checked the test cases and I found that the core logic has already been covered by existing test cases. Openrouter reuses the `customModelName` property to introduce custom models in env and reuses `parseModelString` to parse it, and this flow has been well covered already. There was nothing special about Openrouter. Can yoy be more specific on the additional test coverage you have in mind?

**Comment by arvinxx** [comment]
@danielglh I mean if there is an `openrouterModelString` field, and the result should be snapshot.

**Comment by danielglh** [comment]
> @danielglh I mean if there is an `openrouter?.customModelName` field, and the result should be snapshot.

@arvinxx test case added

File: src/store/global/slices/settings/selectors/modelProvider.test.ts
Comment (arvinxx) [inline src/store/global/slices/settings/selectors/modelProvider.test.ts:1]: Why the chatModels only have openai models?

Comment (danielglh) [reply src/store/global/slices/settings/selectors/modelProvider.test.ts:1]: @arvinxx forgot to enable the model in settings ... fixed

**Review by arvinxx — APPROVED** [review]
LGTM, I'll merge it. Thanks for your contribution!

**Pull request merged**